### ISSUE TITLE
Dropout in GRU/LSTM in Tensorflow 2.0 doesn't reset dropout masks on call

### ISSUE THREAD

**Issue by ptigas** [tensorflow/tensorflow#29391]
**System information**

_Tensorflow_
GIT VERSION: `v1.12.1-3283-geff4ae822a`
VERSION: `2.0.0-dev20190604`

Colab environment

**Describe the current behavior**

In RNNs, the dropout masks should be reset after every call. However, in training mode (where the dropouts are activated) GRU and LSTM implementation in tensorflow 2.0 seems to be re-using the same dropout masks, leading to deterministic behavior.

Simple RNN seems to be doing the right thing, re-sampling dropout masks after each call.

**Describe the expected behavior**

The expected behaviour should be the same as SimpleRNN (re-sample dropout masks on each call).

**Code to reproduce the issue**

The following code produces the correct behaviour in tensorflow 1.13.1:

```
from __future__ import absolute_import, division, print_function

import tensorflow as tf
tf.enable_eager_execution()

import numpy as np

tf.enable_eager_execution()
print(tf.__version__)
data = np.random.normal(0, 1, (1, 10, 2)).astype(np.float32)
rnn = tf.keras.layers.GRU(units=10, dropout=0.5, recurrent_dropout=0.5)
print(set([rnn(data, training=True).numpy()[0, 0] for _ in range(5)]))
```

output
```
1.13.1
{0.09432551, -0.07633728, 0.03358479, 0.010588642, 0.0}
```

but in tensorflow 2.0 it doesn't
```
from __future__ import absolute_import, division, print_function

import tensorflow as tf
import numpy as np

print(tf.version.GIT_VERSION, tf.version.VERSION)
data = np.random.normal(0, 1, (1, 10, 2)).astype(np.float32)
rnn = tf.keras.layers.GRU(units=10, dropout=0.5, recurrent_dropout=0.5)
print(set([rnn(data, training=True).numpy()[0, 0] for _ in range(5)]))
```
output
```
v1.12.1-3283-geff4ae822a 2.0.0-dev20190604
{-0.14212656}
```

A quick fix is to call `reset_dropout_mask()` and `reset_recurrent_dropout_mask()` between calls however this looks like a breaking change.

```
def fixed_rnn():
  rnn.cell.reset_dropout_mask()
  rnn.cell.reset_recurrent_dropout_mask()
  return rnn(data, training=True).numpy()[0, 0]

print(set([fixed_rnn() for _ in range(5)]))
```
output
```
{-0.004232532, 1.1669009, -0.009177759, 3.0901778, 4.5860972}
```

**Comment by achandraa** [comment]
Have tried on Colab with TF CPU versions 1.13.1 and 2.0.0-dev20190604 and was able to reproduce the issue.

**Comment by qlzh727** [comment]
Thanks for reporting the issue. There was a similar issue, and has been recently fixed in 180f28a26660ca2e1ba27477f4f9592db5f9c4e8. Can u try with the latest nightly again?

**Comment by qlzh727** [comment]
https://github.com/tensorflow/tensorflow/issues/29187

**Comment by ptigas** [comment]
For version `v1.12.1-3447-g5a0f1bbfb7 2.0.0-dev20190606`  the issue seems resolved.

**Comment by qlzh727** [comment]
Thanks for verifying this, closing bug now.

**Comment by sbagroy986** [comment]
Thanks a ton for your help on this!

Quick follow-up: has this been fixed in the GPU version as well? I tried the version from yesterday and it didn't seem to work.

**Comment by qlzh727** [comment]
@sbagroy986, the CPU and GPU implementation is covered under same code path, which means they should all be covered by the change. Can u report a issue with details if you feel this hasn't be fixed? Also, please make sure you install the latest pip package, which should include the fix.
